### ISSUE TITLE
schema command panicked at a certain parquet file

### ISSUE THREAD

**Issue by hangxie** [hangxie/parquet-tools#185]
@erikburgess reported that certain parquet file will panick schema command with:

```
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/alecthomas/kong.catch(0x14000559e60)
	github.com/alecthomas/kong@v0.2.16/kong.go:383 +0xb4
panic({0x104e45100, 0x140001e56e0})
	runtime/panic.go:884 +0x204
github.com/hangxie/parquet-tools/cmd.(*schemaNode).updateTagFromConvertedType(0x14000524240, 0x140001128d0?)
	github.com/hangxie/parquet-tools/cmd/schema.go:262 +0x838
github.com/hangxie/parquet-tools/cmd.(*schemaNode).getTagMap(0x14000524240)
	github.com/hangxie/parquet-tools/cmd/schema.go:337 +0x518
github.com/hangxie/parquet-tools/cmd.(*schemaNode).jsonSchema(0x14000524240)
	github.com/hangxie/parquet-tools/cmd/schema.go:127 +0x24
github.com/hangxie/parquet-tools/cmd.(*schemaNode).jsonSchema(0x14000114630)
	github.com/hangxie/parquet-tools/cmd/schema.go:145 +0x160
github.com/hangxie/parquet-tools/cmd.(*SchemaCmd).Run(0x105559d68, 0x1?)
	github.com/hangxie/parquet-tools/cmd/schema.go:37 +0x298
reflect.Value.call({0x104d3d540?, 0x105559d68?, 0x140004d7ac8?}, {0x104ab8de6, 0x4}, {0x1400000fd10, 0x1, 0x1042ef39c?})
	reflect/value.go:584 +0x688
reflect.Value.Call({0x104d3d540?, 0x105559d68?, 0x140004d7b68?}, {0x1400000fd10?, 0xc?, 0xc?})
	reflect/value.go:368 +0x90
github.com/alecthomas/kong.callMethod({0x104ab890b, 0x3}, {0x104dcb920?, 0x105559d68?, 0x3?}, {0x104d3d540?, 0x105559d68?, 0x0?}, 0x0?)
	github.com/alecthomas/kong@v0.2.16/callbacks.go:71 +0x3f0
github.com/alecthomas/kong.(*Context).RunNode(0x140001c0a00, 0x140003aa700, {0x140004d7f00, 0x1, 0x1})
	github.com/alecthomas/kong@v0.2.16/context.go:706 +0x460
github.com/alecthomas/kong.(*Context).Run(0x140003aa1c0?, {0x140004d7f00?, 0x0?, 0x0?})
	github.com/alecthomas/kong@v0.2.16/context.go:723 +0xbc
main.main()
	github.com/hangxie/parquet-tools/main.go:40 +0x2b8
```

**Comment by hangxie** [comment]
This is from v1.15.1, original parquet file cannot be shared due to some reason, it seems raw format of schema works fine, but not go or JSON format.

**Comment by hangxie** [comment]
Confirmed parquet-go's tools failed with similar error, the original parquet file works fine with Java based parquet tools like https://hub.docker.com/r/nathanhowell/parquet-tools.

**Comment by hangxie** [comment]
It seems the offending element is:

```
    {
      "repetition_type": "OPTIONAL",
      "name": "ContractNumbers",
      "num_children": 1,
      "converted_type": "LIST",
      "logicalType": {
        "LIST": {}
      },
      "children": [
        {
          "type": "BYTE_ARRAY",
          "repetition_type": "REPEATED",
          "name": "Array",
          "converted_type": "UTF8",
          "logicalType": {
            "STRING": {}
          }
        }
      ]
    },
```

I don't think I knew this kind of format before, LIST under my expectation should be like:
```
{
  "repetition_type": "REQUIRED",
  "name": "List",
  "num_children": 1,
  "converted_type": "LIST",
  "children": [
    {
      "repetition_type": "REPEATED",
      "name": "List",
      "num_children": 1,
      "children": [
        {
          "type": "BYTE_ARRAY",
          "type_length": 0,
          "repetition_type": "REQUIRED",
          "name": "Element",
          "converted_type": "DECIMAL",
          "scale": 2,
          "precision": 10,
          "field_id": 0,
          "logicalType": {
            "DECIMAL": {
              "scale": 2,
              "precision": 10
            }
          }
        }
      ]
    }
  ]
}
```

Note there is an intermediate level of `List` in my expectation.

**Comment by hangxie** [comment]
Quick fix is that if `Children` has `Type` then I should not expect the intermediate level, I will do a quick fix for this.

**Comment by hangxie** [comment]
Fixed and released v1.15.2
